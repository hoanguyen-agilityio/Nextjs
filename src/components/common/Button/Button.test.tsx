import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import ButtonCustom from '.';

describe('Button component', () => {
  test('renders with default type', () => {
    const { container } = render(
      <ButtonCustom color="default" size="xs" radius="sm">
        Button
      </ButtonCustom>,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders with secondary type', () => {
    const { container } = render(
      <ButtonCustom color="secondary" size="s" radius="md">
        Button
      </ButtonCustom>,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders with primary type', () => {
    const { container } = render(
      <ButtonCustom color="secondary" size="md">
        Button
      </ButtonCustom>,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders with info type', () => {
    const { container } = render(
      <ButtonCustom color="info" size="l">
        Button
      </ButtonCustom>,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders with dark type', () => {
    const { container } = render(
      <ButtonCustom color="dark" size="xl">
        Button
      </ButtonCustom>,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders with grey type', () => {
    const { container } = render(
      <ButtonCustom color="grey" size="xxl">
        Button
      </ButtonCustom>,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders with bright type', () => {
    const { container } = render(
      <ButtonCustom color="bright">Button</ButtonCustom>,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders with unstyled type', () => {
    const { container } = render(
      <ButtonCustom color="unstyled" size="unstyled">
        Button
      </ButtonCustom>,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders the disabled button', () => {
    render(<ButtonCustom isDisabled={true}>Disabled Button</ButtonCustom>);
    const button = screen.getByRole('button', { name: /Disabled Button/i });

    expect(button).toBeDisabled();
    expect(button).toHaveClass('opacity-50 cursor-not-allowed');
  });
});
