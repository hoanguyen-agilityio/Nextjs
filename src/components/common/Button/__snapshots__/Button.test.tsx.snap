// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders with default type 1`] = `
<div>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover bg-white border-translucent-700 dark:border-gray-700 dark:bg-dark-900 dark:text-white py-4 pl-5 pr-4 text-xsm font-semibold border rounded-lg"
    type="button"
  >
    Button
  </button>
</div>
`;
